msu/utilities/type_checkers.nut

```diff
@@ -0,0 +1,67 @@
+::MSU.requireString <- function( ... )
+{
+	::MSU.requireTypeArray("string", vargv);
+}
+
+::MSU.requireInt <- function( ... )
+{
+	::MSU.requireTypeArray("integer", vargv);
+}
+
+::MSU.requireArray <- function( ... )
+{
+	::MSU.requireTypeArray("array", vargv);
+}
+
+::MSU.requireFloat <- function( ... )
+{
+	::MSU.requireTypeArray("float", vargv);
+}
+
+::MSU.requireBool <- function( ... )
+{
+	::MSU.requireTypeArray("bool", vargv);
+}
+
+::MSU.requireTable <- function( ... )
+{
+	::MSU.requireTypeArray("table", vargv);
+}
+
+::MSU.requireInstance <- function( ... )
+{
+	::MSU.requireTypeArray("instance", vargv);
+}
+
+::MSU.requireFunction <- function( ... )
+{
+	::MSU.requireTypeArray("function", vargv);
+}
+
+::MSU.requireType <- function( _type, ... )
+{
+	::MSU.requireTypeArray(_type, vargv);
+}
+
+// Private
+::MSU.requireTypeArray <- function( _type, _values )
+{
+	foreach (value in _values)
+	{
+		if (typeof value != _type)
+		{
+			throw ::MSU.Exception.InvalidType;
+		}
+	}
+}
+
+::MSU.requireOneOfType <- function( _typeArray, ... )
```

Comment:
I'd suggest renaming this to `requireOneFromTypes` rather than `requireOneOfType` because the latter implies that we have a list of variables and we want at least one of those variables to be of some given type. Whereas what this function does is it checks if any one of the given types is present in the given variables.

Reply:
I agree